--- a/system/core/libziparchive/zip_writer.cc
+++ b/system/core/libziparchive/zip_writer.cc
@@ -389,7 +389,7 @@
   CHECK(z_stream_->avail_out != 0);
 
   // Prepare the input.
-  z_stream_->next_in = reinterpret_cast<const uint8_t*>(data);
+  z_stream_->next_in = const_cast<Bytef*>(reinterpret_cast<const Bytef*>(data));
   z_stream_->avail_in = len;
 
   while (z_stream_->avail_in > 0) {
--- a/system/core/libutils/include/utils/Vector.h
+++ b/system/core/libutils/include/utils/Vector.h
@@ -264,11 +264,6 @@
     return *this;
 }
 
-template<class TYPE> inline
-const Vector<TYPE>& Vector<TYPE>::operator = (const Vector<TYPE>& rhs) const {
-    VectorImpl::operator = (static_cast<const VectorImpl&>(rhs));
-    return *this;
-}
 
 template<class TYPE> inline
 Vector<TYPE>& Vector<TYPE>::operator = (const SortedVector<TYPE>& rhs) {
